units/patch_intent.go

```diff
@@ -339,7 +335,9 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		}
 	}
 
-	if len(patchDoc.VariantsTasks) == 0 {
+	skipForFailed := failedOnly && !(previousPatchStatus == evergreen.PatchFailed)
```

Comment:
more succinct would be previousPatchStatus != evergreen.PatchFailed, although I don't really understand the purpose of this.

Reply:
done